### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by hex0cter**
Rewrite the library as the previous is buggy and sometimes cannot find emails.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,41 +1,64 @@
...
+        const id = message.attributes.uid;
+        const idHeader = "Imap-Id: " + id + "\r\n";
+
+        const mail = await parse(idHeader + all.body);
 
-    notifier
-      .on('mail', (mail) => {
         let matched = true;
-        if (from && !mail.from.some((element) => element.address === from)) {
+        if (from && from != mail.from.text) {
```
Comment (cattermo) [inline index.js]:
```suggestion
        if (from && from !== mail.from.text) {
```

File: index.js [hunk 2]
```diff
@@ -1,41 +1,64 @@
...
           matched = false;
         }
-        if (to && !mail.to.some((element) => element.address === to)) {
+
+        if (to && to != mail.to.text) {
```
Comment (cattermo) [inline index.js]:
```suggestion
        if (to && to !== mail.to.text) {
```

File: index.js [hunk 3]
```diff
@@ -1,41 +1,64 @@
...
           matched = false;
         }
-        if (subject && !mail.subject.match(new RegExp(subject))) {
+
+        if (subject && subject != mail.subject) {
```
Comment (cattermo) [inline index.js]:
```suggestion
        if (subject && subject !== mail.subject) {
```

**Review by cattermo — APPROVED** [review]
I suggest using !== instead of != 

Thank you for doing this!

**Comment by hex0cter** [comment]
> I suggest using !== instead of !=
> 
> Thank you for doing this!

Fixed.

**Pull request merged**